src/storage/src/source/persist_source.rs

```diff
@@ -190,3 +194,218 @@ where
 
     (ok_stream, err_stream, token)
 }
+
+pub fn persist_source_listen_split<G>(
```

Comment:
ditto let's replace the old codepaths, the new way is strictly better! concretely: replace `persist_source`'s implementation with this (but keep the name `persist_source`) plz

Reply:
ditto on follow-up PR